app/src/App.vue

```diff
@@ -47,16 +49,42 @@ onBeforeMount(async () => {
         Sentry.captureException(err);
     }
 });
+
+// Wait 1 second to allow the socket connection to be established before checking the connection status
+setTimeout(() => {
+    watch(
+        isConnected,
+        () => {
+            if (!isConnected.value) {
+                useNotificationStore().addNotification({
+                    id: "offlineBanner",
+                    title: "You are offline",
+                    description:
+                        "Do not worry - you can still use the app and browse through the offline content.",
+                    state: "warning",
+                    type: "banner",
+                    icon: SignalSlashIcon,
+                });
+            }
+
+            if (isConnected.value) {
+                useNotificationStore().removeNotification("offlineBanner");
+            }
+        },
+        { immediate: true },
+    );
+}, 1000);
 </script>
 
 <template>
     <TopBar class="sticky top-0 z-40" />
+    <NotificationManager type="banner" class="sticky" />
```

Comment:
This generates a warning in the console:

```
[Vue warn]: Extraneous non-props attributes (class) were passed to component but could not be automatically inherited because component renders fragment or text root nodes. 
```

Basically, you shouldn't pass the class here, but put it on the `div` in the component itself. An extra reason to split up the NotificationManager in two.